File: engine/model/src/main/scala/tools/packet/foreigneffect/ShowHint.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowHint(private var _hint: String, private var _width: Int, private var _height: Int) extends PacketInput {
  def hint: String = _hint

  def width: Int = _width

  def height: Int = _height

  override def opcode(): SendOpcode = SendOpcode.PLAYER_HINT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowHint.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowHint(String hint, Integer width, Integer height) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_HINT;
   }
}